### PR TITLE
Defer loading youtube script and image optimizations for home page

### PR DESCRIPTION
**Pull request by zchase**

### CODE REVIEW

File: layouts/index.html [hunk 1]
```diff
@@ -230,7 +230,13 @@ <h3>{{ .Params.video.heading }}</h3>
...
         <div class="rounded shadow-md" style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
-            <iframe src="//www.youtube.com/embed/6f8KF6UGN7g?rel=0" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" allowfullscreen="" title="Introduction to Pulumi: Modern Infrastructure as Code"></iframe>
+            <iframe
+                src="//www.youtube.com/embed/6f8KF6UGN7g?rel=0"
+                style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;"
+                allowfullscreen=""
+                title="Introduction to Pulumi: Modern Infrastructure as Code"
+                srcdoc="<style>*{padding:0;margin:0;overflow:hidden}html,body{height:100%}img,span{position:absolute;width:100%;top:0;bottom:0;margin:auto}span{height:1.5em;text-align:center;font:48px/1.5 sans-serif;color:white;text-shadow:0 0 0.5em black}</style><a href=https://www.youtube.com/embed/6f8KF6UGN7g?autoplay=1><img src=https://img.youtube.com/vi/6f8KF6UGN7g/hqdefault.jpg alt='Introduction to Pulumi: Modern Infrastructure as Code'><span>▶</span></a>"
```
Comment (zchase) [inline layouts/index.html:238]: @cnunciato What do you think of this? We should get a cleaner image for the preview if we go this route, but it makes a huge difference in page speed. Once I did all the image optimizations we were getting ~48 for the page speed and once I added this `srcdoc` attribute we got up to ~70.

File: layouts/index.html [hunk 2]
```diff
@@ -119,27 +119,27 @@ <h1 class="font-sans font-body font-normal">
                     <img class="mx-auto lg:mx-8 h-12 mx-8" src="/logos/customers/tableau_logo.png" alt="Tableau Software" title="Tableau Software">
                 </a>
                 <a data-track="customer-logo" href="{{ relref . "/case-studies/mercedes-benz" }}">
-                    <img class="mx-auto lg:mx-8 mt-8 mb-4 lg:my-0 h-12 mx-8" src="/logos/customers/mercedes-benz-RDNA_logo.png" title="Mercedes-Benz Research and Development" alt="Mercedes-Benz Research and Development">
+                    <img class="mx-auto lg:mx-8 mt-8 mb-4 lg:my-0 h-12 mx-8" src="/logos/customers/logo-section/mercedes-benz-RDNA_logo.png" title="Mercedes-Benz Research and Development" alt="Mercedes-Benz Research and Development">
```
Comment (cnunciato) [inline layouts/index.html:122]: Maybe submit these changes as a separate PR, since they're unrelated?

Comment (zchase) [reply layouts/index.html:122]: They're related because the images are a bit big in the logo section and Google Page Speed was complaining. I didn't want to edit the main logo images to avoid messing up any other place they're used. Though I can break it out into two PRs.

**Comment by zchase** [comment]
@cnunciato I took a swing at generating a new image for the video preview. This should be all good now

**Comment by cnunciato** [comment]
@zchase The play button is pretty big and it's obscuring the text. Any chance you could drop it down to something about 2X the height of the capital letters, and add some transparency like the YouTube button does?

image

Feel free to pass me the source image if you'd rather not mess with this, though.

File: layouts/index.html (hunk 2 above)
Comment (cnunciato) [reply layouts/index.html:122]: Were you planning on doing this? (Breaking it into a separate PR, I mean.)

I will say that if the images really are are "big", since they're PNGs and of a fixed size, so you should just optimize them in place, and not create copies of them. If you grep for them you should find anywhere else they're referenced. But I suspect it should be fine to do this.

**Comment by zchase** [comment]
Pre merge page speeds for https://pulumi.com/

developers google com_speed_pagespeed_insights__url=https%3A%2F%2Fpulumi com%2F tab=mobile (1)
developers google com_speed_pagespeed_insights__url=https%3A%2F%2Fpulumi com%2F tab=mobile

**Comment by zchase** [comment]
Post Merge Page Speeds:

developers google com_speed_pagespeed_insights__url=https%3A%2F%2Fpulumi com%2F tab=mobile (3)
developers google com_speed_pagespeed_insights__url=https%3A%2F%2Fpulumi com%2F tab=mobile (2)
